airbyte-cdk/python/airbyte_cdk/utils/schema_inferrer.py

```diff
@@ -84,11 +84,12 @@ def _clean(self, node: InferredSchema):
                     node["type"] = [node["type"], "null"]
                     node.pop("anyOf")
             if "properties" in node:
-                for key, value in list(node["properties"].items()):
-                    if isinstance(value, dict) and value.get("type", None) == "null":
-                        node["properties"].pop(key)
-                    else:
-                        self._clean(value)
+                if isinstance(node["properties"], dict):
```

Comment:
this should always be true, but mypy doesn't know. Better safe than sorry

Reply:
nit: would it be more readable to have `if "properties" in node and isinstance(node["properties"], dict):`?